src/XR/webXRSessionManager.ts

```diff
@@ -337,16 +337,69 @@ export class WebXRSessionManager implements IDisposable {
             });
     }
 
+    /** @hidden */
+    public _addLayerWrapperProvider(provider: WebXRLayerWrapperProvider) {
+        this._layerWrapperProviders.add(provider);
+    }
+
+    /** @hidden */
+    public _removeLayerWrapperProvider(provider: WebXRLayerWrapperProvider) {
+        this._layerWrapperProviders.delete(provider);
+    }
+
+    private _updateXRLayers(layers: XRLayer[]) {
+        // Wrap the new layers
+        this._xrLayers = layers.map(
+            (layer: XRLayer) => {
+                let layerWrapper: Nullable<WebXRLayerWrapper> = null;
+                // We assume that the order doesn't matter and either
+                // only one provider is capable of producing a wrapper for a given layer
+                // or multiple providers would give the same answer.
+                this._layerWrapperProviders.forEach((provider) => {
+                    if (!layerWrapper) {
+                        layerWrapper = provider.createLayerWrapper(layer);
+                    }
+                });
+                // If all else fails just assume it's a regular old XRWebGLLayer.
+                if (!layerWrapper) {
+                    layerWrapper = WebXRLayerWrapper.CreateFromXRWebGLLayer(layer as XRWebGLLayer);
+                }
+                return layerWrapper;
+            });
+
+        this._baseRTTProvider?.dispose();
+        this._baseRTTProvider = this._xrLayers[0].createRenderTargetProvider(this);
+    }
+
     /**
-     * Updates the render state of the session
+     * Updates the render state of the session.
+     * Note that this is deprecated in favor of webXRSessionManager.updateRenderState().
      * @param state state to set
      * @returns a promise that resolves once the render state has been updated
      */
-    public updateRenderStateAsync(state: XRRenderState) {
-        if (state.baseLayer) {
-            this._baseLayer = state.baseLayer;
+    public updateRenderStateAsync(state: XRRenderState): Promise<void> {
+        if (!!state.baseLayer) {

```

Comment:
I don't think so, this is just a style that I adopted a long time ago because I read a pedantic javascript article-- is there a recommendation for using/avoiding `!!` in Babylon.js?

Reply:
i see. `scene.baseLayer` will be "truthy", so no need to convert it to a boolean (which is what `!!` does). Sometimes typescript complains because it expects a boolean (depends on the scenario, of course).
But this is not that important :-)